src/blockchain/esplora/reqwest.rs

```diff
@@ -167,9 +167,27 @@ impl Blockchain for EsploraBlockchain {
                         .request()
                         .map(|txid| {
                             let tx = tx_index.get(txid).expect("must be in index");
-                            (tx.previous_outputs(), tx.to_tx())
+                            // Verify this transaction if requested via feature flag
+                            #[cfg(feature = "verify")]
+                            {
+                                use crate::wallet::verify::VerifyError;
+                                let prev_outs = tx.previous_outputs();
+                                let tx_bytes = serialize(&tx.to_tx());
+                                for (index, output) in prev_outs.iter().enumerate() {
+                                    if let Some(output) = output {
+                                        bitcoinconsensus::verify(
+                                            output.script_pubkey.to_bytes().as_ref(),
+                                            output.value,
+                                            &tx_bytes,
+                                            index,
+                                        )
+                                        .map_err(|e| VerifyError::from(e))?;
+                                    }
+                                }
+                            }
+                            Ok((tx.previous_outputs(), tx.to_tx()))
```

Comment:
I think we could just do this in `script_sync` and be done with it? I get that we want to use `script_sync` logic for rpc soon which doesn't need to verify but we can just turn this off at runtime with a flag when we start the sync.

Reply:
Yes this makes more sense. I will try to see where I can fit this in `script_sync`.